package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1799,6 +1800,137 @@ def n_fragments(self):
 
+    def unwrap(self, box=None, compound='fragments', reference='com',
+               inplace=True):
+        """Move atoms in this group so that bonds within the
+        group's compounds aren't split accross periodic boundaries.
+
+        Parameters
+        ----------
+        box : array_like or None, optional
+            The unitcell dimensions of the system, which can be orthogonal or
+            triclinic and must be provided in the same format as returned by
+            :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:\n
+            ``[lx, ly, lz, alpha, beta, gamma]``.
+            If ``None``, uses the dimensions of the current timestep.
+        compound : {'group', 'segments', 'residues', 'molecules', 'fragments'}, optional
+            Which type of components to unwrap. Note that, in any case, all
+            atoms within each compound must be interconnected by bonds, i.e.,
+            compounds must correspond to (parts of) molecules.
+        reference : {'com', 'cog', None}, optional
+            If ``'com'`` (center of mass) or ``'cog'`` (center of geometry), the
+            unwrapped compounds will be shifted so that their individual
+            reference point lies within the primary unit cell.
+            If ``None``, no such shift is performed.
+        inplace : bool, optional
+            If ``True``, coordinates are modified in place.
+
+        Returns
+        -------
+        coords : numpy.ndarray
+            Unwrapped atom coordinate array of shape ``(n, 3)``.
+
+        Note
+        ----
+        * This is the inverse transformation of packing atoms into the primary
+          unit cell, preventing bonds from being "broken" accross periodic
+          boundaries.
+        * Be aware of the fact that only atoms *belonging to the group* will
+          be unwrapped! If you want entire molecules to be unwrapped, make sure
+          that all atoms of these molecules belong the group.
+
+        See Also
+        --------
+        :func:`MDAnalysis.lib.mdamath.make_whole`
+
+
+        .. versionadded:: 0.20.0
+        """
+        if reference is not None:
+            if reference.lower() not in ['com', 'cog']:
+                raise ValueError("Unrecognized reference definition: {}\n"
+                                 "Please use one of 'com', 'cog', or None."
+                                 "".format(compound))
+            ref = reference.lower()
+
+        if box is None:
+            box = self.dimensions
+        boxtype, box = check_box(box)
+        ortho = (boxtype == 'ortho')
+        if ortho:
+            hbox = 0.5 * box[:3]
+
+        atoms = self.atoms
+        unique_atoms = atoms.unique
+
+        comp = compound.lower()
+        if comp == 'group':
+            if ortho:
+                positions = unique_atoms.positions
+                # Only unwrap if necessary:
+                if np.any(np.abs(positions - positions[0]) > hbox):
```

Comment:
this optimisation is cute, but maybe deserves to live inside of `make_whole`

Reply:
Very true! That way, the optimization will always be applied.

Reply:
also if it's done in the Cython level you can exit the `any()` loop early, I think this will do the whole calculation then evaluate the boolean values

Reply:
For an atomgroup consisting of 500 fragments with 25 atoms per fragment, i measured the following runtimes:
* without optimization: 154 ms,
* optimization in `make_whole()`: 65 ms
* optimization in Python: 66.5 ms

Turns out the speed-up is not that great anymore. This simple loop runs in 22 ms on the same system:
```Python
def group_make_whole(ag):
    box = ag.dimensions
    fragments = ag.fragments
    if np.any(box[3:] != 90.0):
        for f in fragments:
            mda.lib.mdamath.make_whole(f)
    else:
        hbox = 0.5 * box[:3]
        for f in fragments:
            pos = f.positions
            if np.any(np.abs(pos - pos[0]) > hbox):
                mda.lib.mdamath.make_whole(f)
```
Seems to be the rest of the machinery that slows down `group.unwrap()`.